src/main/java/org/opensearch/plugin/insights/rules/model/Measurement.java

```diff
@@ -0,0 +1,220 @@
+        } else if (a instanceof Float && b instanceof Float) {
+            return a.floatValue() + b.floatValue();
+        } else {
+            throw new IllegalArgumentException("Unsupported number type: " + a.getClass() + " or " + b.getClass());
+        }
+    }
+
+    /**
+     * Get measurement number based on the aggragation type
+     * @return measurement number
+     */
+    public Number getMeasurement() {
+        switch (aggregationType) {
+            case NONE:
+            case SUM:
+                return number;
+            case AVERAGE:
+                return getAverageMeasurement(number, count);
+            default:
+                throw new IllegalArgumentException("Aggregation Type should be set for measurement.");
+        }
+    }
+
+    /**
+     * Get average measurement number based on the total and count
+     * @param total total measurement value
+     * @param count count of measurements
+     * @return average measurement value
+     */
+    private Number getAverageMeasurement(Number total, int count) {
+        if (count == 0) {
+            throw new IllegalArgumentException("Count cannot be zero for average calculation.");
+        }
+
+        if (total instanceof Long) {
+            return ((Long) total) / count;
+        } else if (total instanceof Integer) {
+            return ((Integer) total) / count;
+        } else if (total instanceof Double) {
+            return ((Double) total) / count;
+        } else if (total instanceof Float) {
+            return ((Float) total) / count;
+        } else {
+            throw new IllegalArgumentException("Unsupported number type: " + total.getClass());
+        }
+    }
+
+    /**
+     * Set measurement
+     * @param measurement measurement number
+     */
+    public void setMeasurement(Number measurement) {
+        number = measurement;
+    }
+
+    /**
+     * Set aggregation type
+     * @param aggregationType aggregation type
+     */
+    public void setAggregationType(AggregationType aggregationType) {
+        this.aggregationType = aggregationType;
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, ToXContent.Params params) throws IOException {
+        builder.startObject();
+        builder.field("number", number);
+        builder.field("count", count);
+        builder.field("aggregationType", aggregationType.toString());
+        builder.endObject();
+        return builder;
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        writeNumber(out, number);
+        out.writeInt(count);
+        out.writeString(aggregationType.toString());
+    }
+
+    private void writeNumber(StreamOutput out, Number number) throws IOException {
+        if (number instanceof Long) {
+            out.writeByte((byte) 0); // Type indicator for Long
+            out.writeLong((Long) number);
+        } else if (number instanceof Integer) {
```

Comment:
Why do we need to implement the `Writeable` interface? Probably can get rid of this if not really needed